src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -260,6 +260,56 @@ public TypeSymbolWithAnnotations SetIsAnnotated(CSharpCompilation compilation)
         public TypeSymbolWithAnnotations AsNullableReferenceType() => _extensions.AsNullableReferenceType(this);
         public TypeSymbolWithAnnotations AsNotNullableReferenceType() => _extensions.AsNotNullableReferenceType(this);
 
+        public (bool, TypeSymbolWithAnnotations) TryMergeNullability(TypeSymbolWithAnnotations other, VarianceKind variance)
+        {
+            (bool isNullableSuccess, bool? isNullable) = TryMergeIsNullable(IsNullable, other.IsNullable, variance);
+            if (isNullableSuccess)
+            {
+                (bool typeSuccess, TypeSymbol type) = TypeSymbol.TryMergeNullability(other.TypeSymbol, variance);
+                if (typeSuccess)
+                {
+                    return (true, Create(type, isNullable, CustomModifiers));
+                }
+            }
+            return (false, default);
+        }
+
+        private static (bool, bool?) TryMergeIsNullable(bool? a, bool? b, VarianceKind variance)
```

Comment:
>private [](start = 8, length = 7)

Please document the returned value (I think: a pair, the first of which indicates if the caller should report a warning, the second of which is the computed merged nullability)